src/main/java/org/isf/medicalstock/service/MovementIoOperationRepository.java

```diff
@@ -59,9 +59,13 @@ public interface MovementIoOperationRepository extends JpaRepository<Movement, I
 	@Query(value = "select max(mov.date) from Movement mov")
 	LocalDateTime findMaxDate();
 
-	@Query(value = "select mov.refNo from Movement mov where mov.refNo like :refNo")
-	List<String> findAllWhereRefNo(@Param("refNo") String refNo);
+	@Query(value = "select mov from Movement mov where mov.refNo = :refNo")
+	List<Movement> findAllWhereRefNo(@Param("refNo") String refNo);
+	
+	@Query(value = "SELECT * FROM OH_MEDICALDSRSTOCKMOV ORDER BY MMV_ID DESC limit 1", nativeQuery = true)
```

Comment:
Why not like this? 

```suggestion
	@Query(value = "select m from Movement order by m.code DESC limit 1")
```

Reply:
Why you markred as resolved without applying suggestions or without saying why you won't apply it?

Reply:
when I apply the changes the tests fails. i think it is about the limit clause.

Reply:
the limit clause is in both statements, so what do you mean?

Reply:
ok, looks like "limit" is a vendor dependent function so it has not been implemented in JPA/Hiberanate, I didn't expected that.
There are workarounds for that, but let's leave it as is, thanks.

Reply:
okay